dojson/contrib/to_marc21/fields/bd01x09x.py

```diff
@@ -1364,7 +1382,27 @@ def reverse_national_agricultural_library_copy_statement(self, key, value):
 @utils.filter_values
 def reverse_subject_category_code(self, key, value):
     """Reverse - Subject Category Code."""
+
+    indicator_map2 = {
+        "Source specified in subfield $2": "7",
+        "NAL subject category code list ": "0"
+    }
+
+    field_map = {
+        'subject_category_code': 'a',
+        'subject_category_code_subdivision': 'x',
+        'source': '2',
+        'linkage': '6',
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[4] in indicator_map2:
+        order.append('code_source')
```

Comment:
Something like that (I guess you mean if key[4] **!=** '7'):
```
@to_marc21.over('072', '^subject_category_code$')
@utils.reverse_for_each_value
@utils.filter_values
def reverse_subject_category_code(self, key, value):
    """Reverse - Subject Category Code."""

    indicator_map2 = {
        "Source specified in subfield $2": "7",
        "NAL subject category code list ": "0"
    }

    field_map = {
        'subject_category_code': 'a',
        'subject_category_code_subdivision': 'x',
        'source': '2',
        'linkage': '6',
        'field_link_and_sequence_number': '8',
    }

    order = utils.map_order(field_map, value)

    if key[4] in indicator_map2:
        order.append('code_source')

    if key[4] != '7':
        order.remove('2')

    return {
        '__order__': tuple(order) if len(order) else None,
        'a': value.get('subject_category_code'),
        '8': utils.reverse_force_list(
            value.get('field_link_and_sequence_number')
        ),
        '2': value.get('source'),
        'x': utils.reverse_force_list(
            value.get('subject_category_code_subdivision')
        ),
        '6': value.get('linkage'),
        '$ind1': '_',
        '$ind2': indicator_map2.get(value.get('code_source'), '_'),
    }
```
(double checking cause the more I work with it, the more I get confused)